packages/teleterm/src/services/tshd/createStartupClient.ts

```diff
@@ -0,0 +1,59 @@
+import grpc from '@grpc/grpc-js';
+
+import { StartupServiceClient } from 'teleterm/services/tshd/v1/startup_service_grpc_pb';
+import * as api from 'teleterm/services/tshd/v1/startup_service_pb';
+import Logger from 'teleterm/logger';
+
+import middleware, { withLogging } from './middleware';
+
+export function createStartupClient(
+  addr: string,
+  credentials: grpc.ChannelCredentials
+) {
+  const logger = new Logger('startup');
+  const grpcClient = middleware(new StartupServiceClient(addr, credentials), [
+    withLogging(logger),
+  ]);
+
+  return {
+    async resolveTshdEventsServerAddress(address: string) {
```

Comment:
I see how it can be confusing.

It's a tough one. This RPC resolves the address on tshd side. So you have StartupService there with `ResolveTshdEventsServerAddress`, the renderer process is a client here which submits some info to the tshd, playing the role of a server here. So technically we can think of it as creating some sort of a resource on the server.

I mentioned that there's already an issue with how I designed the backend https://github.com/gravitational/webapps/pull/1285#issuecomment-1289180091. I suppose I could rename it to `UpdateTshdEventsServerAddress` which in turn would always set up a new tshd events client on tshd side.